components/edit_channel_purpose_modal/edit_channel_purpose_modal.jsx

```diff
@@ -114,13 +87,13 @@ export default class EditChannelPurposeModal extends React.Component {
     }
 
     handleSave = () => {
-        const {channel, actions: {patchChannel}} = this.props;
+        const {channel, actions: {getChannel, patchChannel}} = this.props;
         const {purpose} = this.state;
         if (!channel) {
             return;
         }
 
-        patchChannel(channel.id, {purpose});
+        patchChannel(channel.id, {purpose}).then(() => getChannel(channel.id));
```

Comment:
Do we still need to get the channel after patching something that isn't the header? Shouldn't the client not replace the previous props after a patch?

If we need to do this all the time when we patch the channel, we may want to change the API to always return the channel with the new props set

Reply:
See below, but I had been treating the header and purpose as both formattable fields.

You're right: we should be returning the updated channel in the patch itself. In fact, things are already wired up to do that, but I failed to make the patch endpoint actually fill in the channel props. I'll fix that!